plugins/vocabbuilder.koplugin/main.lua

```diff
@@ -0,0 +1,1180 @@
+            self.got_it_button,
+            self.more_button,
+        }
+        table.insert(self.layout, self.forgot_button)
+        table.insert(self.layout, self.got_it_button)
+        table.insert(self.layout, self.more_button)
+    else
+        local star = Button:new{
+            icon = "check",
+            icon_width = star_width,
+            icon_height = star_width,
+            bordersize = 0,
+            radius = 0,
+            margin = 0,
+            show_parent = self,
+            enabled = false,
+            no_focus = true,
+        }
+        right_side_width =  Size.padding.large * 3 + self.item.review_count * (star:getSize().w)
+
+        if self.item.review_count > 0 then
+            right_widget = HorizontalGroup:new {
+                dimen = Geom:new{w=0, h = self.height}
+            }
+            for i=1, self.item.review_count, 1 do
+                table.insert(right_widget, star)
+            end
+        else
+            star:free()
+            right_widget = HorizontalGroup:new{
+                dimen = Geom:new{w=0, h = self.height},
+                 HorizontalSpan:new {width = Size.padding.default }
+            }
+        end
+        table.insert(right_widget, self.margin_span)
+        table.insert(right_widget, self.more_button)
+        table.insert(self.layout, self.more_button)
+    end
+
+    local text_max_width = self.width - point_widget_width - right_side_width
+
+    local subtitle_prefix = TextWidget:new{
+        text = BD.mirroredUILayout() and self:getTimeSinceDue() .. _("From ") or self:getTimeSinceDue() .. _("From ") ,
+        face = subtitle_face,
+        fgcolor = subtitle_color
+    }
+
+    self[1] = FrameContainer:new{
+        padding = 0,
+        bordersize = 0,
+        HorizontalGroup:new{
+            dimen = Geom:new{
+                w = self.width,
+                h = self.height,
+            },
+            HorizontalGroup:new{
+                dimen = Geom:new{
+                    w = self.width - right_side_width,
+                    h = self.height,
+                },
+                VerticalGroup:new{
+                    dimen = Geom:new{w = point_widget_width, h = self.height},
+                    self.point_v_spacer,
+                    point_widget,
+                    VerticalSpan:new { width = self.height - point_widget_height - self.point_v_spacer.width}
+                },
+                VerticalGroup:new{
+                    dimen = Geom:new{
+                        w = text_max_width,
+                        h = self.height,
+                    },
+                    self.v_spacer,
+                    LeftContainer:new{
+                        dimen = Geom:new{w = text_max_width, h = word_height},
+                        word_widget
+                    },
+                    LeftContainer:new{
+                        dimen = Geom:new{w = text_max_width, h = self.height - word_height - self.v_spacer.width*2.2},
```

Comment:
That one would also fall into the 'needs to be truncated to an integer' basket ;).